### PR TITLE
xbps-src: generate python pkgs for multiple versions automatically

### PR DESCRIPTION
**Pull request by asergi**
The patch has been successfully tested on several python modules.

Any suggestion or help is appreciated.

### CODE REVIEW

File: common/xbps-src/shutils/pkgtarget.sh [hunk 1]
```diff
@@ -99,6 +100,19 @@ remove_pkg() {
             msg_normal "$f: removing 32bit files from destdir...\n"
             rm -rf ${_destdir}/${f}-32bit-${version}
         fi
+        for pyver in $python_versions; do
```
Comment (Vaelatern) [inline common/xbps-src/shutils/pkgtarget.sh]: Wouldn't a better way to deal with this be to optionally add the `python${pyver}-${pymodn}` to the for loop this is in, to do the full processing on it, perhaps to the `${subpackages}` variable that they currently get processed in?

File: Manual.md [hunk 2]
```diff
@@ -1055,6 +1055,11 @@ by blanks, i.e `pycompile_module="foo blah"`.
 recursively by the target python version. This differs from `pycompile_module` in that any
 path may be specified, i.e `pycompile_dirs="usr/share/foo"`.
 
+- `py2depends`: this variable expects python modules that are only
+needed by the Python 2 version of the package. These are often
+backports of Python 3.x standard library modules for use with Python 2,
+e.g.: `py2depends="python-enum34 python-ipaddress"`.
+
 > NOTE: it's expected that additional subpkgs must be generated to allow packaging for multiple
```
Comment (Vaelatern) [inline Manual.md]: This PR would remove this `NOTE`

File: Manual.md [hunk 3]
```diff
@@ -1055,6 +1055,11 @@ by blanks, i.e `pycompile_module="foo blah"`.
 recursively by the target python version. This differs from `pycompile_module` in that any
 path may be specified, i.e `pycompile_dirs="usr/share/foo"`.
 
+- `py2depends`: this variable expects python modules that are only
```
Comment (Vaelatern) [inline Manual.md]: `this variable expects dependencies that are only`

File: common/hooks/do-pkg/00-gen-pkg.sh [hunk 4]
```diff
@@ -129,6 +130,60 @@ hook() {
 		_replaces+=" ${_pkgn}-32bit${_pkgv}"
 	done
 
+	# Generate pythonX.Y-foo pkg.
+	if [ -z "$python_versions" -o "$python_versions" = "${python_versions#2.7}" ]; then
+		return
+	fi
+	for pyver in $python_versions; do
+		if [ "$pyver" != "2.7" ]; then
```
Comment (Vaelatern) [inline common/hooks/do-pkg/00-gen-pkg.sh]: Should also loop over `depends`, checking for dependencies and automatically changing them to the python `pyver` version of that dependency, if such a thing exists. Alternately a new variable for python 3 specific depends, such as used in the `python-PyQt4` template.

**Comment by Gottox** [comment]
I'm not sure if this is a good idea. This is a lot of added complexity that will work for python only. So the process of building a python template differs from ruby, lua, ...

Until now we had a very simple rule:

> One package, one directory (or link) in void-packages/srcpkgs
> (except for dbg and 32bit which are in its own repository)

This PR breaks this rule, makes it harder to understand how a package is built and adds special treatment for one script language only.

**Comment by Vaelatern** [comment]
Except python 3 subpackages are not an unknown value. It's a lot of retyping of stuff that is relatively basic to pull off a python 3 subpackage.

**Comment by asergi** [comment]
Obsolete. We switched to python3 packages, instead.

**Pull request closed without merging**